# Data Analysis - Lab 3
### M. Sébastien MASCHA & M. Sauvage Pierre
### ISEP Paris – September 24th, 2019
<br/>
<br/>

___
# Import of libraries

This document has been done using python on Jupyter Notebook with the librairies:
 
- maths for sqrt, pi, exp
- Numpy to manipulate arrays
- pandas to import csv
- matplotlib to plot graphics
- seaborn to make your charts prettier (built on top of Matplotlib)
- sklearn : tools for data mining and data analysis
- mlxtend : tools for ploting PCA

In [4]:
# coding: utf-8

import data

from math import sqrt,pi,exp
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns; sns.set()

import sklearn
# Normalize data
from sklearn.preprocessing import StandardScaler
# Dimension reduction
from sklearn.decomposition import PCA
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis as QDA
# Useful 
from sklearn.linear_model import LogisticRegression

# Exercice B - MDS, LLE and Isomap
We are now going to study the "golub" data. These data contains the level of expression of 7129 genes based on 72 sample. Each sample is linked to a variant of Leukemia: AML(25) and ALL(47). Our goal is to visualize the 72 samples on a 2D plan.
#### Question 1 - Open golub_data.csv as a matrice

We use the comma separator because we saw in the text editor that the data was separated
by semicolon.

In [9]:
df= pd.read_csv('data/golub_data.csv', sep=",")
print(df.shape)

(7129, 72)


#### Question 5 - Compute the confidence intervals for the correlation coefficients (we will suppose that the attributes are following a normal distribution). Comment your results.

In [42]:
def IC95(a,b):
    r = pearsonr(a,b)[0]
    Z = np.log(abs(1 + r)) - np.log(abs(1 - r))/2
    sz = sqrt(1/(len(a) - 3))
    Zinf = Z - 1.96*sz
    Zsup = Z + 1.96*sz
    ic_interval = [(exp(2*Zinf) - 1)/(exp(2*Zinf) + 1), (exp(2*Zsup) - 1)/(exp(2*Zsup) + 1)]
    return ic_interval  

# Calcul des 16 intervalles de confiance à 95% :
for i in range(dataframe.shape[1]):
    for j in range(dataframe.shape[1]):
        print(IC95(dataframe.values[:,i],dataframe.values[:,j]))

[nan, nan]
[0.13426981777053296, 0.41938373810452045]
[0.06035688449301633, 0.3559920722670827]
[0.1550204052207662, 0.4366943233822553]
[0.1751305926080555, 0.4532739744948389]
[0.16815644198526442, 0.4475459190283694]
[0.0887412499451348, 0.38066213211057265]
[0.10693257323975644, 0.39625746160692854]
[-0.2588919772266451, 0.046903597018188345]
[0.13426981777053296, 0.41938373810452045]
[nan, nan]
[0.6389565094017534, 0.7888073290213987]
[0.9197188757950654, 0.9561567588569714]
[0.8466097523006582, 0.9147568587731572]
[0.9295649109392146, 0.9616232266107839]
[0.8370486424958964, 0.9092347028400082]
[0.6746510962043446, 0.8113917905064472]
[0.032076467069970915, 0.33099559346489804]
[0.06035688449301633, 0.3559920722670827]
[0.6389565094017534, 0.7888073290213987]
[0.9999999999999997, 1.0]
[0.5544844868835288, 0.7336987196236355]
[0.46038938471837654, 0.6694047245794875]
[0.6044785482041218, 0.7666015712819692]
[0.16368253010302636, 0.4438592656734183]
[0.6481288671477571, 0.794649695

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: RuntimeWarning: divide by zero encountered in log
  This is separate from the ipykernel package so we can avoid doing imports until


It appears that there is a strong link between the height and the length of the femur as well as with the span of the arm which have a coefficient of determination of around 0.8.

There is also a 0.64 and 0.62 coefficient of determination between the height and the size of the feet as well as with the length of the hand, making them less strongly linked.

We observe about a 0.55 determination between the length of the femur and the size of the feet as well as with the length of the hand. It shows are strong they're linked.

Furthermore, penis' size is definitely not linked with attributes provided.